oxo.py

```diff
@@ -1,26 +1,130 @@
 class OxoBoard:
     def __init__(self):
         """ The initialiser. Initialise any fields you need here. """
-        raise NotImplementedError("TODO: implement __init__")
+        self.number_to_win_for = 3
+        self.rows = 3
+        self.cols = 3
+
+        self.numberOfTiles = self.rows * self.cols
+
+        # Create and initialise the game board
+        self.board = []
+        for row in range(self.rows):
+            self.board += [[0] * self.cols]
+        print self.board
 
     def get_square(self, x, y):
         """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+        return self.board[y][x]
 
     def set_square(self, x, y, mark):
         """ If the specified square is currently empty (0), fill it with mark and return True.
             If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+        
+        if self.get_square(x, y) == 0:
+            self.board[y][x] = mark
+            return True
+        else:
+            return False
 
     def is_board_full(self):
         """ If there are still empty squares on the board, return False.
             If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+
+        # number of tiles on the board
+        tiles_checked = 0
+
+        # goes through each tile and checks if they are filled or not
+        for row in self.board:
+            for item in row:
+                if item != 0:
+                    tiles_checked += 1
+                    if tiles_checked == self.numberOfTiles:
+                        return True
 
     def get_winner(self):
         """ If a player has three in a row, return 1 or 2 depending on which player.
             Otherwise, return 0. """
-        raise NotImplementedError("TODO: implement get_winner")
+
+        # Player score for each row check
+        p1_count = 0
+        p2_count = 0
+
+        # item/row counters
+        v_row = 0
+        d_item = 0
+        d_item2 = 2
+
+        # check diagonal row
+        for row in self.board:
+            if row[d_item] == 1:
+                p1_count += 1
```

Comment:
Storing the counts in a list or dictionary could have eliminated a lot of these `if` statements. However do you really need separate counts for players 1 and 2 if you're just checking for a line?